web/ui/react-app/src/pages/Services.tsx

```diff
@@ -2,14 +2,113 @@ import React, { FC } from 'react';
 import { Alert } from 'reactstrap';
+import { useFetch } from '../utils/useFetch';
+import { LabelsTable } from './LabelsTable';
 
-const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => (
-  <>
-    <h2>Service Discovery</h2>
-    <Alert color="warning">
-      This page is still under construction. Please try it in the <a href={`${pathPrefix}/service-discovery`}>Classic UI</a>.
-    </Alert>
-  </>
-);
+interface DiscoveredLabels {
+  address: string;
+  metrics_path: string;
+  scheme: string;
+  job: string;
+  my: string;
+  your: string;
+}
+
+interface ActiveTargets {
+  discoveredLabels: DiscoveredLabels[];
+  labels: any;
+  scrapePool: string;
+  scrapeUrl: string;
+  lastError: string;
+  lastScrape: string;
+  lastScrapeDuration: number;
+  health: string;
+}
+
+interface ServiceMap {
+  activeTargets: ActiveTargets[];
+  droppedTargets: any[];
+}
+
+const Services: FC<RouteComponentProps & PathPrefixProps> = ({ pathPrefix }) => {
+  const { response, error } = useFetch<ServiceMap>(`${pathPrefix}/api/v1/targets`);
+  const processTargets = (response: ServiceMap) => {
+    console.warn(response);
+    const activeTargets = response.activeTargets;
+    const targets: any = {};
+
+    // Get targets of each type along with the total and active end points
+    for (const target of activeTargets) {
+      const { scrapePool: name } = target;
+      if (!targets[name]) {
+        targets[name] = {
+          total: 0,
+          active: 0,
+        };
+      }
+      targets[name].total++;
+      targets[name].active++;
+    }
+    for (const target of response.droppedTargets) {
+      const { scrapePool: name } = target;
+      targets[name].total++;
+    }
+
+    return targets;
+  };
+
+  const processLabels = (response: ActiveTargets[]) => {
+    const labels: any = {};
+    const activeTargets = response;
+
+    for (const target of activeTargets) {
+      const name = target.scrapePool;
+      if (!labels[name]) {
+        labels[name] = [];
+      }
+      labels[name].push({
+        discoveredLabels: target.discoveredLabels,
+        labels: target.labels,
+      });
+    }
+
+    return labels;
+  };
+  let targets: any;
+  let labels: any;
+
+  if (error) {
+    return (
+      <Alert color="danger">
+        <strong>Error:</strong> Error fetching Service-Discovery: {error.message}
+      </Alert>
+    );
+  } else if (response.data) {
+    targets = processTargets(response.data);
+    labels = processLabels(response.data.activeTargets);
+
+    return (
+      <>
+        <h2>Service Discovery</h2>
+        <ul>
+          {Object.keys(targets).map((val, i) => (
+            <li key={i}>
+              <a href={'#' + val}>
+                {' '}
+                {val} ({targets[val].active} / {targets[val].total} active targets){' '}
+              </a>
+            </li>
+          ))}
+        </ul>
+        <hr />
+        {Object.keys(labels).map((val: any, i) => {
```

Comment:
This table only shows active targets (if I fix the crash bug for when there's a dropped target). It also needs to show information about dropped targets, like the old UI:

![dropped](https://user-images.githubusercontent.com/538008/71322229-9e3a4e80-24c5-11ea-8a34-0fd4b6621366.png)

You can introduce a dropped target for testing by adding the following scrape config to your Prometheus config:

```
  - job_name: 'testjob'
    static_configs:
      - targets: # Targets to probe through the Blackbox Exporter.
        - 'keep-this:8080'
        - 'drop-this:8080'
    relabel_configs:
      - source_labels: [__address__]
        regex: 'drop-this:8080'
        action: drop
```